cdap-app-fabric/src/main/java/co/cask/cdap/internal/provision/ProvisionerConfig.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.provision;
+
+import com.google.gson.JsonElement;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Objects;
+
+/**
+ * Contains the config for the provisioner
+ */
+public class ProvisionerConfig {
+  @SerializedName("configuration-groups")
+  private final JsonElement configurationGroups;
```

Comment:
does this work if it's a `List<Object>`? Would like to avoid having gson specific classes as variables.